book/build-service.md

```diff
@@ -0,0 +1,142 @@
+# Build Service
+
+
+## Overview
+
+The Build Service is composed of controllers that create and configure build pipelines. The main input for the Build Service is a Component CR managed by the [Hybrid Application Service (HAS)](hybrid-application-service.md).
+
+![](../diagrams/build-service/build-service.drawio.svg)
+
+### Dependencies
+
+The Build Service is dependent on the following services:
+- [Pipeline Service](./pipeline-service.md)
+  - Pipeline execution, Pipeline logging
+- [Hybrid Application Service](./hybrid-application-service.md)
+  - Provides Component CR with annotations and `.status.devfile` which is used for PipelineRun configuration.
+- [Image Controller](./image-controller.md)
+  - Generation of container image repository which is used by PipelineRun
+
+## Controllers
+
+The Build Service contains these controllers:
+- Component Build Controller
+  - Used for PipelineRuns creation.
+- PaC PipelineRun Pruner Controller
+  - Deletes PipelineRuns managed by Pipelines As Code for Components that were deleted.
+- Git Tekton Resources Renovater
+  - Provides updates for the .tekton directory in the user repository which has been created by Component Build Controller in Custom Mode.
+
+### Component Build Controller
+
+Component Build Controller is managed by Component CR changes. It's using Component CR annotations and `.status.devfile` for selecting a mode and configuration of the PipelineRuns.
+
+#### Modes
+
+Component Build Controller is working in two modes:
+- Default Mode
+  - Mode where PipelineRuns are created directly by the Build Service.
+- Custom Mode
+  - Integrates with Pipelines-As-Code, the Component Build Controller creates PipelineRuns definitions in the user code repository.
```

Comment:
Probably we should use a unified way to describe "Pipelines as Code" and avoid using PaC in this document.

Pipelines as Code website uses Pipelines-as-Code, and from my point of view, "Pipelines as Code" is ok as well. Whichever is selected, just use it consistently.